### PR TITLE
Enable to delete a product

### LINKED ISSUE #21: A seller should be able to delete a specific item
### Description

The seller should be able to delete a specific item in their collection such that they can remove items they no longer deal in or were erroneously added. This functionality ensures that sellers can manage their product listings effectively and maintain an accurate inventory.

### Acceptance Criteria

- Implement a delete option for each item in the seller's inventory.
- Ensure only authenticated sellers can access the delete functionality.
- When a seller selects delete, prompt them to confirm the deletion.
- If the seller confirms, delete the item and display a success message.
- If the seller cancels, no changes are made to the inventory.
- If the item does not exist, display an error message.
- If deletion fails due to an error, display an error message and keep the item in the inventory.

### Development Notes:

- Implement checks to ensure only authenticated sellers can delete items.
- Design a clear interface with delete options and a confirmation dialog for deletion.
-  Ensure user-friendly error messages are displayed for all potential failure scenarios.

### Definition of Done:

- Sellers can see and use the delete option for each item in their inventory.
- Deletion requires seller confirmation to proceed.
- Successfully deleted items display a confirmation message.
- Canceled deletions do not alter the inventory.
- Errors, such as attempting to delete a non-existent item or encountering deletion failures, are appropriately managed and communicated to the seller.

### PR DESCRIPTION
**Pull request by UwicyezaG**
### Description

The seller should be able to delete a specific item in their collection such that they can remove items they no longer deal in or were erroneously added. This functionality ensures that sellers can manage their product listings effectively and maintain an accurate inventory.

### Definition of Done:

- Sellers can see and use the delete option for each item in their inventory.
- Deletion requires seller confirmation to proceed.
- Successfully deleted items display a confirmation message.
- Canceled deletions do not alter the inventory.
- Errors, such as attempting to delete a non-existent item or encountering deletion failures, are appropriately managed and communicated to the seller.

#### Screenshots
Screenshot (13)
Screenshot (11)
Screenshot (15)

### CODE REVIEW

**Review by MC-Knight — APPROVED** [review]

**Review by faid-terence — APPROVED** [review]
@UwicyezaG 
Please rebase with develop and resolve the conflicts 
And increase the test coverage !!

**Review by faid-terence — CHANGES_REQUESTED** [review]
@UwicyezaG
Please rebase with develop and resolve the conflicts
And increase the test coverage !!

**Review by maxCastro1 — APPROVED** [review]

**Review by Ndevu12 — APPROVED** [review]

File: src/Routes/Router.tsx [hunk 1]
```diff
@@ -26,17 +26,17 @@ const Router = () => {
   const { userToken } = useSelector((state: RootState) => state.auth);
   const { decodedToken } = useJwt<DecodedToken>(userToken);
 
-  const isAdmin = decodedToken?.role.toLowerCase() === 'admin';
-  const isVendor = decodedToken?.role.toLowerCase() === 'vendor';
-  const isBuyer = decodedToken?.role.toLowerCase() === 'buyer';
+  const isAdmin = decodedToken?.userType.toLowerCase() === 'admin';
```
Comment (aimedivin) [inline src/Routes/Router.tsx]: @UwicyezaG  You must use the value from 'role' instead of 'userType', as the token does not contain userType.

**Review by aimedivin — CHANGES_REQUESTED** [review]

File: src/components/Products/DashboardSingleProduct/DashboardSingleProduct.tsx
Comment (faid-terence) [inline src/components/Products/DashboardSingleProduct/DashboardSingleProduct.tsx:1]: @UwicyezaG Are we using userType or role ?

**Review by faid-terence — CHANGES_REQUESTED** [review]

**Comment by faid-terence** [comment]
@UwicyezaG 
Work on the change requests and rebase with develop to resolve the conflicts

**Review by GSinseswa721 — APPROVED** [review]

**Review by elijahladdie — APPROVED** [review]

**Review by faid-terence — APPROVED** [review]
PR Approved @UwicyezaG !

**Pull request merged**